aspnetcore/mvc/controllers/filters.md

```diff
@@ -86,216 +82,224 @@ Several of the filter interfaces have corresponding attributes that can be used
+> Every controller that inherits from the `Controller` base class includes `OnActionExecuting` and `OnActionExecuted` methods. These methods wrap the filters that run for a given action:  `OnActionExecuting` is called before any of the filters, and `OnActionExecuted` is called after all of the filters.
 
-To modify the default, scope-based order, you could explicitly set the `Order` property of a class-level or method-level filter. For example, adding `Order=-1` to a method level attribute:
-
-```csharp
-[MyFilter(Name = "Method Level Attribute", Order=-1)]
-   ```
-
-In this case, a value of less than zero would ensure this filter ran before both the Global and Class level filters (assuming their `Order` property was not set).
-
-The new order would be:
-
-1. The Controller `OnActionExecuting`
+## Authorization Filters
 
-2. The Method filter `OnActionExecuting`
+*Authorization Filters* control access to action methods and are the first filters to be executed within the filter pipeline. They have only a before method, unlike most filters that support before and after methods. You should only write a custom authorization filter if you are writing your own authorization framework. Note that you should not throw exceptions within authorization filters, since nothing will handle the exception (exception filters won't handle them). Instead, issue a challenge or find another way.
 
-3. The Global filter `OnActionExecuting`
+Learn more about [Authorization](../../security/authorization/index.md).
 
-4. The Class filter `OnActionExecuting`
+## Resource Filters
 
-5. The Class filter `OnActionExecuted`
+*Resource filters* implement either the `IResourceFilter` or `IAsyncResourceFilter` interface, and their execution wraps most of the filter pipeline. (Only [Authorization filters](#authorization-filters) run before them.) Resource filters are especially useful if you need to short-circuit most of the work a request is doing. For example, a caching filter can avoid the rest of the pipeline if the response is already in the cache.
 
-6. The Global filter `OnActionExecuted`
+The [short circuiting resource filter](#short-circuiting-resource-filter) shown earlier is one example of a resource filter. A very naive cache implementation (do not use this in production) that only works with `ContentResult` action results is shown below:
 
-7. The Method filter `OnActionExecuted`
+[!code-csharp[Main](./filters/sample/src/FiltersSample/Filters/NaiveCacheResourceFilterAttribute.cs?name=snippet_CacheFilter&highlight=1,2,11,16,17,27,30)]
 
-8. The Controller `OnActionExecuted`
+In this example, `OnResourceExecuting` checks if the result is already in the static dictionary cache. If so, it sets the `Result`, which short-circuits the rest of the pipeline. In the `OnResourceExecuted` method, if the current request's key isn't already in use, the current `Result` is stored in the cache.
```

Comment:
And sorry I don't have a more constructive example we can use here, we've implemented all of the useful things. If you're not averse to linking to another sample (or including it) https://github.com/aspnet/Entropy/blob/dev/samples/Mvc.FileUpload/Filters/DisableFormValueModelBindingAttribute.cs

This is a filter that will prevent model binding from accessing the form data. It's useful for cases where you know that you're going to receive large file uploads and want to prevent the form from being read into memory.